src/group.rs

```diff
@@ -32,52 +42,100 @@ impl GroupCreateOpts {
         id: Option<GroupId>,
         name: Option<GroupName>,
+        add_as_admin: bool,
         add_as_member: bool,
+        owner: Option<UserId>,
+        admins: Vec<UserId>,
         members: Vec<UserId>,
         needs_rotation: bool,
     ) -> GroupCreateOpts {
         GroupCreateOpts {
             id,
             name,
+            add_as_admin,
             add_as_member,
+            owner,
+            admins,
             members,
             needs_rotation,
         }
     }
 
-    fn standardize(self, calling_id: &UserId) -> GroupCreateOpts {
+    pub(crate) fn standardize(self, calling_id: &UserId) -> Result<GroupCreateOptsStd> {
+        // if `add_as_member`, make sure the calling user is in the `members` list
         let standardized_members = if self.add_as_member && !self.members.contains(calling_id) {
             let mut members = self.members.clone();
             members.push(calling_id.clone());
             members
         } else {
             self.members
         };
-        GroupCreateOpts::new(
-            self.id,
-            self.name,
-            self.add_as_member,
-            standardized_members,
-            self.needs_rotation,
-        )
+        let (standardized_admins, owner_id) = {
+            // if `add_as_admin`, make sure the calling user is in the `admins` list
+            let mut admins = if self.add_as_admin && !self.admins.contains(calling_id) {
+                let mut admins = self.admins.clone();
+                admins.push(calling_id.clone());
+                admins
+            } else {
+                self.admins
+            };
+            let owner: &UserId = match &self.owner {
+                Some(owner_id) => {
+                    // if the owner is specified, make sure they're in the `admins` list
+                    if !admins.contains(owner_id) {
+                        admins.push(owner_id.clone());
+                    }
+                    owner_id
+                }
+                // if the owner is the calling user (default), they should have been added to the
+                // admins list by `add_as_admin`. If they aren't, it will error later on.
+                None => calling_id,
+            };
+            (admins, owner)
+        };
+
+        if !standardized_admins.contains(owner_id) {
+            Err(IronOxideErr::ValidationError(
+                format!("admins"),
+                format!("admins list must contain the owner"),
+            ))
+        } else {
+            Ok(GroupCreateOptsStd {
+                id: self.id,
+                name: self.name,
+                owner: self.owner,
+                admins: standardized_admins,
+                members: standardized_members,
+                needs_rotation: self.needs_rotation,
+            })
+        }
     }
 }
 
 impl Default for GroupCreateOpts {
+    // Default GroupCreateOpts for common use cases. The user who calls `group_create()` will be the owner of the group
```

Comment:
If these have `///` they _might_ show up on the doc site.

Reply:
Changed